From 53a70289c2712808e6d4e88927e03cac01b470dd Mon Sep 17 00:00:00 2001
From: Bram Moolenaar <Bram@vim.org>
Date: Mon, 9 May 2022 13:15:07 +0100
Subject: [PATCH] patch 8.2.4925: trailing backslash may cause reading past end
 of line

Problem:    Trailing backslash may cause reading past end of line.
Solution:   Check for NUL after backslash.
---
 src/testdir/test_textobjects.vim | 10 +++++++++-
 src/textobject.c                 |  4 ++++
 src/version.c                    |  2 ++
 3 files changed, 15 insertions(+), 1 deletion(-)

diff --git a/src/textobject.c b/src/textobject.c
index e4a7db38e..edaa64c51 100644
--- a/src/textobject.c
+++ b/src/textobject.c
@@ -1664,7 +1664,11 @@ find_next_quote(
 	if (c == NUL)
 	    return -1;
 	else if (escape != NULL && vim_strchr(escape, c))
+	{
 	    ++col;
+	    if (line[col] == NUL)
+		return -1;
+	}
 	else if (c == quotechar)
 	    break;
 	if (has_mbyte)
-- 
2.36.1

